### PR TITLE
LUCENE-10195: Add gradle cache option and make some tasks cacheable

### PR DESCRIPTION
**Pull request by dweiss**

### CODE REVIEW

**Comment by uschindler** [comment]
Wasn't the idea to make the cache optional? Looks like it is enabled by default.

File: gradle/documentation/render-javadoc.gradle [hunk 1]
```diff
@@ -284,8 +300,14 @@ class RenderJavadocTask extends DefaultTask {
   @Input
   boolean relativeProjectLinks = false
 
-  @Input
-  def offlineLinks = [:]
+  @Internal
+  Map<String, File> offlineLinks = [:]
+
+  // Computes cacheable inputs from the map in offlineLinks.
+  @Nested
```
Comment (jprinet) [inline gradle/documentation/render-javadoc.gradle:307]: nice job

File: gradle/generation/local-settings.gradle [hunk 2]
```diff
@@ -55,6 +55,11 @@ org.gradle.jvmargs=-Xmx3g \\
 org.gradle.parallel=true
 org.gradle.priority=normal
 
+# This setting enables local task output caches. This will speed up
+# your local builds in most cases but will also consume disk space in your
+# gradle home. See LUCENE-10195 and/or SOLR-15603 for details.
+# org.gradle.caching=true
```
Comment (jprinet) [inline gradle/generation/local-settings.gradle:61]: I think it's good to have it commented out and documented that way 👍

**Review by jprinet — APPROVED** [review]
LGTM 👌

**Pull request merged**